### PR TITLE
feat(puppeteer): update 1.8 and improve typings

### PR DESCRIPTION
**Pull request by SimonSchick**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/GoogleChrome/puppeteer/releases https://github.com/GoogleChrome/puppeteer/blob/v1.8.0/docs/api.md
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

- Added changes from 1.7 and 1.8.
- Refactor docs to use `@default` correctly.
- Fixed `pageerror` event being declared as `string`.
- Use overloads for `waitFor` to improve type safety.
- Use `Metrics` interface for `metrics` event.
- Narrowed `env` parameter.
- Narrowed `CDPSession.send` return type.

### CODE REVIEW

File: types/puppeteer/index.d.ts [hunk 1]
```diff
@@ -473,6 +474,13 @@ export interface NavigationOptions {
    * @default load Navigation is consider when the `load` event is fired.
    */
   waitUntil?: LoadEvent | LoadEvent[];
+
+  /**
+   * Referer header value.
+   * If provided it will take preference over the referer header value set by
+   * [page.setExtraHTTPHeaders()](#pagesetextrahttpheadersheaders).
+   */
+  referer?: string;
```
Comment (marvinhagemeister) [inline types/puppeteer/index.d.ts]: This interface is used for `goBack(options)` which doesn't have this property according to the  api docs: https://github.com/GoogleChrome/puppeteer/blob/v1.8.0/docs/api.md#pagegobackoptions

Comment (SimonSchick) [reply types/puppeteer/index.d.ts]: That's annoying, I will create a new interface then.

File: types/puppeteer/index.d.ts [hunk 2]
```diff
@@ -1258,7 +1330,7 @@ export interface Page extends EventEmitter, FrameBase {
   deleteCookie(...cookies: DeleteCookie[]): Promise<void>;
 
   /** Emulates given device metrics and user agent. This method is a shortcut for `setUserAgent` and `setViewport`.  */
-  emulate(options: Partial<EmulateOptions>): Promise<void>;
+  emulate(options: EmulateOptions): Promise<void>;
```
Comment (marvinhagemeister) [inline types/puppeteer/index.d.ts]: Can you confirm that the all properties must be present? Last time I checked the api docs where  a bit vague on that front.

Comment (SimonSchick) [reply types/puppeteer/index.d.ts]: All properties in `EmulateOptions` are optional already, so the `Partial` monad was redundant.

**Comment by SimonSchick** [comment]
Uhm, so I got a conflict between my local lint and travis lint...

**Review by marvinhagemeister — APPROVED** [review]
Awesome, thanks for updating the PR. LGTM on my part 🎉

**Pull request merged**